zyte_spider_templates/heuristics.py

```diff
@@ -56,3 +63,43 @@ def might_be_category(url: str) -> bool:
                 return False
 
     return True
+
+
+INDEX_URL_PATHS = {
+    "",
+    "/index",
+    "/index.html",
+    "/index.htm",
+    "/index.php",
+    "/home",
+}
+
+
+def is_homepage(url: str) -> bool:
+    """Given a URL, returns True if the URL could be a homepage."""
+    url_split = urlsplit(url)
+    url_path = url_split.path.rstrip("/").lower()
+
+    # Finds and removes URL subpaths like "/us/en", "en-us", "en-uk", etc.
+    if _url_has_locale_pair(url_path):
+        url_path = url_path[6:]
+
+    # Finds and removes URL subpaths like "/en", "/fr", etc.
+    match = re.search(r"/(\w{2})(?!\w)", url_path)
+    if match and (match.group(1) in LANG_CODES or match.group(1) in COUNTRY_CODES):
+        url_path = url_path[3:]
```

Comment:
I’m curious if this is more efficient than building a regular expression out of `LANG_CODES` and `COUNTRY_CODES` instead.

Reply:
Good thought there. I've benchmarked this and found out that they're the same.